pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -176,7 +177,7 @@ func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *mode
 			t := true
 			_, err := c.Patch(context.Background(), name, types.ApplyPatchType, data, metav1.PatchOptions{
 				Force:        &t,
-				FieldManager: constants.ManagedGatewayController,
+				FieldManager: controllerName,
```

Comment:
Don't we also need the following changes?
```
diff --git a/pilot/pkg/config/kube/gateway/deploymentcontroller.go b/pilot/pkg/config/kube/gateway/deploymentcontroller.go
index da910e9153..e24e501fd0 100644
--- a/pilot/pkg/config/kube/gateway/deploymentcontroller.go
+++ b/pilot/pkg/config/kube/gateway/deploymentcontroller.go
@@ -122,7 +122,7 @@ var builtinClasses = getBuiltinClasses()
 
 func getBuiltinClasses() map[gateway.ObjectName]gateway.GatewayController {
        res := map[gateway.ObjectName]gateway.GatewayController{
-               defaultClassName:                 constants.ManagedGatewayController,
+               defaultClassName:                 controllerName,
                constants.RemoteGatewayClassName: constants.UnmanagedGatewayController,
        }
        if features.EnableAmbientControllers {
```
```
diff --git a/pilot/pkg/config/kube/gateway/deploymentcontroller.go b/pilot/pkg/config/kube/gateway/deploymentcontroller.go
index e24e501fd0..5c86f6941f 100644
--- a/pilot/pkg/config/kube/gateway/deploymentcontroller.go
+++ b/pilot/pkg/config/kube/gateway/deploymentcontroller.go
@@ -122,7 +122,7 @@ var builtinClasses = getBuiltinClasses()
 
 func getBuiltinClasses() map[gateway.ObjectName]gateway.GatewayController {
        res := map[gateway.ObjectName]gateway.GatewayController{
-               defaultClassName:                 controllerName,
+               defaultClassName:                 constants.ManagedGatewayController,
                constants.RemoteGatewayClassName: constants.UnmanagedGatewayController,
        }
        if features.EnableAmbientControllers {
@@ -133,8 +133,8 @@ func getBuiltinClasses() map[gateway.ObjectName]gateway.GatewayController {
 
 func getClassInfos() map[gateway.GatewayController]classInfo {
        m := map[gateway.GatewayController]classInfo{
-               constants.ManagedGatewayController: {
-                       controller:         constants.ManagedGatewayController,
+               controllerName: {
+                       controller:         controllerName,
                        description:        "The default Istio GatewayClass",
                        templates:          "kube-gateway",
                        defaultServiceType: corev1.ServiceTypeLoadBalancer,
```
If I correctly understand the reconciliation logic (lines 254-283), custom gateway classes will never match the controller name set in `FieldManager`:
```
// Reconcile takes in the name of a Gateway and ensures the cluster is in the desired state
func (d *DeploymentController) Reconcile(req types.NamespacedName) error {
	log := log.WithLabels("gateway", req)

	gw := d.gateways.Get(req.Name, req.Namespace)
	if gw == nil {
		log.Debugf("gateway no longer exists")
		// we'll ignore not-found errors, since they can't be fixed by an immediate
		// requeue (we'll need to wait for a new notification), and we can get them
		// on deleted requests.
		return nil
	}

	var controller gateway.GatewayController
	var gc *gateway.GatewayClass
	if d.gatewayClasses != nil {
		gc = d.gatewayClasses.Get(string(gw.Spec.GatewayClassName), "")
	}
	if gc != nil {
		controller = gc.Spec.ControllerName
	} else {
		if builtin, f := builtinClasses[gw.Spec.GatewayClassName]; f {
			controller = builtin
		}
	}
	ci, f := classInfos[controller]
	if !f {
		log.Debugf("skipping unknown controller %q", controller)
		return nil
	}
```

Reply:
Yes, makes sense. Thanks. Addressed with fixup to commit [0f5f150](https://github.com/maistra/istio/pull/1002/commits/0f5f15019aa7848b6cc4430bd3142ae44c873705).